builtin/providers/aws/resource_aws_iam_user.go

```diff
@@ -0,0 +1,115 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/service/iam"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsIamUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIamUserCreate,
+		Read:   resourceAwsIamUserRead,
+		// There is an UpdateUser API call, but goamz doesn't support it yet.
+		// XXX but we aren't using goamz anymore.
+		//Update: resourceAwsIamUserUpdate,
+		Delete: resourceAwsIamUserDelete,
+
+		Schema: map[string]*schema.Schema{
+			"arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			/*
+				The UniqueID could be used as the Id(), but none of the API
+				calls allow specifying a user by the UniqueID: they require the
+				name. The only way to locate a user by UniqueID is to list them
+				all and that would make this provider unnecessarilly complex
+				and inefficient. Still, there are other reasons one might want
+				the UniqueID, so we can make it availible.
+			*/
+			"unique_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"path": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "/",
+				ForceNew: true,
+			},
```

Comment:
Noted. The docs I've given you therefore need more work. I'll keep chugging along and let you know in [that PR](https://github.com/bitglue/terraform/pull/1) when I think I've got something valid.